network/network.go

```diff
@@ -235,8 +235,9 @@ func (n *Network) Listen() {
 	}
 }
 
-// Client either creates or returns a cached peer client given its host address.
-func (n *Network) Client(address string) (*PeerClient, error) {
+// initClient either returns a cached peer client or creates a new one given a net.Conn
+// or dials the client if no net.Conn is provided.
+func (n *Network) initClient(address string, conn net.Conn) (*PeerClient, error) {
```

Comment:
might rename this `getOrSetPeerClient()`, init seems like it can only be done once